Sources/ParseSwift/Authentication/3rd Party/ParseSpotify/ParseSpotify.swift

```diff
@@ -0,0 +1,198 @@
+//
+//  ParseSpotify.swift
+//  ParseSwift
+//
+//  Created by Ulaş Sancak on 06/20/22.
+//  Copyright © 2022 Parse Community. All rights reserved.
+//
+
+import Foundation
+
+// swiftlint:disable line_length
+
+/**
+ Provides utility functions for working with Spotify User Authentication and `ParseUser`'s.
+ Be sure your Parse Server is configured for [sign in with Spotify](https://docs.parseplatform.org/parse-server/guide/#spotify-authdata)
+ For information on acquiring Spotify sign-in credentials to use with `ParseSpotify`, refer to [Spotify's Documentation](https://developer.spotify.com/documentation/general/guides/authorization/)
+ */
+public struct ParseSpotify<AuthenticatedUser: ParseUser>: ParseAuthentication {
+
+    /// Authentication keys required for Spotify authentication.
+    enum AuthenticationKeys: String, Codable {
+        case id
+        case accessToken = "access_token"
+        case clientId = "client_id"
+        case expirationDate = "expiration_date"
+        case refreshToken = "refresh_token"
```

Comment:
Thanks for the addition! Can you post the Spotify link for the authentication keys?

Reply:
I’m sorry. What do you mean by post the link? You mean the base api url to later use with the keys?

Reply:
These keys are different than what’s posted on the Parse Server documentation? https://docs.parseplatform.org/parse-server/guide/#oauth-and-3rd-party-authentication

I’m assuming you got them from the Spotify developer documentation can you post the link specifying these keys are what should be used instead of what’s in the Parse documentation?

Reply:
These are the some useful additional keys to use later. This is the guide: https://developer.spotify.com/documentation/general/guides/authorization/code-flow/
Du you want me to add this somewhere?

Reply:
The `clientId` seems like it should be set on the parse server-side auth adapter and not client side. You probably don't want this value hardcoded in your app. See the apple and twitter auth: https://docs.parseplatform.org/parse-server/guide/#apple-authdata

You will probably want to open another PR in the docs to show others how to do this. You can include the spotify link there: https://github.com/parse-community/docs/blob/gh-pages/_includes/parse-server/third-party-auth.md

Reply:
My guess is yes and would probably start here: https://github.com/parse-community/parse-server/blob/4c29d4d23b67e4abaf25803fe71cae47ce1b5957/src/Adapters/Auth/spotify.js. You would need to open an issue and the parse server repo and the server team would be able to answer questions better than me,

You can see the original PR on the server here: https://github.com/ParsePlatform/parse-server/pull/1226

Reply:
I think I am going to revert the client_id implementation since I am not experienced on server part. And it does not have a usage too, it was for a potential use. Thank you anyway. I will keep those in my mind.

Reply:
I reverted the client_id implementation @cbaker6.